### ISSUE TITLE
`wf1-2` - add handling so job does NOT fail if 0 TEIs fetched `wf1-1`

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/openfn-lime#12]
## Describe the bug and expected behavior 
Sometimes zero records will be returned by DHIS2. See example run where date cursor = `2023-06-01T09:43:37.115`
`# of TEIs extracted :: 0`
https://www.openfn.org/projects/msf-lime/runs/064784c3-ea9f-78eb-9de6-7d0f34f02adb

That's OK. But unfortunately, currently `wf1-2` step fails with `TypeError`: https://www.openfn.org/projects/msf-lime/runs/064784c4-1e1c-7ca8-ab10-d407d31e6237

Can you please update `wf1-2` to pass if 0 records are fetched in job 1. You log a message like `No data fetched in step prior to sync.`

Similarly, also make sure `wf1-3` will pass. 

## To Reproduce
1. Re-run `wf1-1` job to test: https://www.openfn.org/projects/msf-lime/runs/064784c3-ea9f-78eb-9de6-7d0f34f02adb

### expression.js
https://github.com/OpenFn/openfn-lime/blob/main/jobs/wf1-2-createPatients.js
https://github.com/OpenFn/openfn-lime/blob/main/jobs/wf1-3-updateTEIs.js

### state.json
DHIS2:
-data: {}
-configuration: see LP MSF DHIS2 Test Org (api user)
-cursor: see above to reproduce 

OpenMRS:
- configuration: see LP MSF OpenMRS Demo

## Toggl 
MSF LIME

**Comment by mtuchi** [comment]
@aleksa-krolls i don't think is caused by the upstream job having `0` TEIs, i think it's the `birthdate` problem again. Because if you look at the logs it says 
```
# of TEIs to send to OpenMRS:  22
TypeError [Error]: Cannot read properties of null (reading '0')
 at vm.js:56:8
```
It mean we have `22` records to send to `OpenMRS`

I will investigate further that line 56 which seems to be causing the problem

**Comment by aleksa-krolls** [comment]
@mtuchi as discussed, to calculate DOB from age, we should set: 
1. dd = current `day` of today()  -->  (so if we run on June 1st, return `01`)
2. MM = current `month` of today() --> (so if we run on June 1st, return `06`)
3. YYYY = current `year` of today() - `age` --> (so if `age=22` and we run on 1 June 2023, return `2001` because 2023 - 22 = 2001)

**Comment by mtuchi** [comment]
@aleksa-krolls i have pushed a PR that will fix this issue See PR 👉🏽 #14
